flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/BufferResponseDecoderDelegate.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.netty;
+
+import org.apache.flink.runtime.io.network.NetworkClientHandler;
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBuf;
+import org.apache.flink.shaded.netty4.io.netty.buffer.ByteBufAllocator;
+
+/**
+ * The parser for {@link org.apache.flink.runtime.io.network.netty.NettyMessage.BufferResponse}.
+ */
+public class BufferResponseDecoderDelegate implements NettyMessageDecoderDelegate {
+
+	/** The network client handler of current channel. */
+	private final NetworkClientHandler networkClientHandler;
+
+	/** The Flink Buffer allocator. */
+	private final NetworkBufferAllocator allocator;
+
+	/** The cumulation buffer of message header. */
+	private ByteBuf messageHeaderCumulationBuffer;
```

Comment:
make it shorter. `messageHeaderCumulationBuffer` -> headerBuf`?

Reply:
I think it might be better to rename to `messageHeaderBuffer` since in concept we call the second part `Message Header`.